promql/functions.go

```diff
@@ -986,29 +1026,30 @@ var functions = map[string]*Function{
 		Call:       funcCountScalar,
 	},
 	"days_in_month": {
-		Name:         "days_in_month",
-		ArgTypes:     []model.ValueType{model.ValVector},
-		OptionalArgs: 1,
-		ReturnType:   model.ValVector,
-		Call:         funcDaysInMonth,
+		Name:       "days_in_month",
+		ArgTypes:   []model.ValueType{model.ValVector},
+		Variadic:   1,
+		ReturnType: model.ValVector,
+		Call:       funcDaysInMonth,
 	},
 	"day_of_month": {
-		Name:         "day_of_month",
-		ArgTypes:     []model.ValueType{model.ValVector},
-		OptionalArgs: 1,
-		ReturnType:   model.ValVector,
-		Call:         funcDayOfMonth,
+		Name:       "day_of_month",
+		ArgTypes:   []model.ValueType{model.ValVector},
+		Variadic:   1,
+		ReturnType: model.ValVector,
+		Call:       funcDayOfMonth,
 	},
 	"day_of_week": {
-		Name:         "day_of_week",
-		ArgTypes:     []model.ValueType{model.ValVector},
-		OptionalArgs: 1,
-		ReturnType:   model.ValVector,
-		Call:         funcDayOfWeek,
+		Name:       "day_of_week",
+		ArgTypes:   []model.ValueType{model.ValVector},
+		Variadic:   1,
+		ReturnType: model.ValVector,
+		Call:       funcDayOfWeek,
 	},
 	"delta": {
 		Name:       "delta",
-		ArgTypes:   []model.ValueType{model.ValMatrix},
+		ArgTypes:   []model.ValueType{model.ValMatrix, model.ValScalar},
+		Variadic:   1, // The 2nd argument is deprecated.
```

Comment:
According to [this](https://github.com/prometheus/prometheus/pull/1147#discussion-diff-41262090) conversation, adding a variadic int field was proposed instead of optionalArgs.